### PR TITLE
Upgrade Ruby version to 3.2.1

### PR DESCRIPTION
**Pull request by pablogrs**
Ruby version upgraded to 3.2.1
Java version upgraded to Java 11 using SDKMan

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by dduportal — COMMENTED** [review]
Thanks for this contribution ? Would you be willing to update your PR to use the JDK embeded in the `jenkins/inbound-agent:alpine` image ?

The Jenkins project uses Temurin and that would be a great way to keep a constant behavior WDYT?

**Comment by pablogrs** [comment]
> Thanks for this contribution ? Would you be willing to update your PR to use the JDK embeded in the `jenkins/inbound-agent:alpine` image ?
> 
> The Jenkins project uses Temurin and that would be a great way to keep a constant behavior WDYT?

Hi, of course, let me check if I understand first.
Instead of using [jenkins-inbound](https://github.com/jenkinsci/docker-inbound-agents/pull/35/files#diff-4b62900532c579e3c7c952ed7e68a5614cfb8f35adf4640c4523c54fd3e29271R1) as a source for Jenkins files you are asking to make it the base image, is that right?

**Comment by dduportal** [comment]
> Hi, of course, let me check if I understand first. Instead of using [jenkins-inbound](https://github.com/jenkinsci/docker-inbound-agents/pull/35/files#diff-4b62900532c579e3c7c952ed7e68a5614cfb8f35adf4640c4523c54fd3e29271R1) as a source for Jenkins files you are asking to make it the base image, is that right?

Both "methods" are valid (each one as its pros and cons)

- Either using the `jenkins/inbound-agent` as base image,
- OR importing only the requirements using the current "multistage" pattern.

In order to avoid "big changes" , what do you think keeping the actual pattern, but copying the requirements from the `jnlp` stage?

You have an example here: https://github.com/jenkins-infra/docker-packaging/blob/7c8ac52f2bfe510d5431ccc7169aa3233700bec2/Dockerfile#L93-L95

File: ruby/Dockerfile [hunk 1]
```diff
@@ -1,13 +1,27 @@
-FROM jenkins/inbound-agent:alpine as jnlp
+FROM jenkins/inbound-agent:alpine as jenkins-agent
```
Comment (dduportal) [inline ruby/Dockerfile:1]:
```suggestion
# jdk11 is a Debian based image
FROM jenkins/inbound-agent:jdk11 as jenkins-agent
```

**Review by dduportal — CHANGES_REQUESTED** [review]
That is an excellent work!

I've added one last mandatory suggestion to use a Debian based Inbound Agent to ensure that the JDK used is prepared for Debian glibc (as the previous Alpine would have a JDK installation for musl which will probably not work).


Once added, the PR is ready to go!

File: ruby/Dockerfile (hunk 1 above)
Comment (pablogrs) [reply ruby/Dockerfile]: Thanks for that, I didn't realise, yes it would make it fail!

**Review by dduportal — APPROVED** [review]
LGTM, many thanks!

**Pull request merged**